plugin.video.vrt.nu/resources/lib/vrtplayer/vrtplayer.py

```diff
@@ -26,140 +17,80 @@ class VRTPlayer:
-        menu_items = self.__get_az_menu_items(joined_url, {'class': 'nui-tile'},
-                                              actions.LISTING_VIDEOS,
-                                              self.metadata_collector.get_az_metadata)
+    def show_tvshow_menu_items(self, path):
+        menu_items = self._api_helper.get_tvshow_items(path)
         self._kodi_wrapper.show_listing(menu_items, sortmethod.ALPHABET)
 
     def show_category_menu_items(self):
-        joined_url = urljoin(self.VRTNU_BASE_URL, './categorieen/')
-        menu_items = self.__get_category_menu_items(joined_url, {'class': 'nui-tile title'}, actions.LISTING_CATEGORY_VIDEOS)
+        joined_url = ''.join((self.VRTNU_BASE_URL, '/categorieen/'))
+        menu_items = self.__get_category_menu_items(joined_url, {'class': 'nui-tile title'}, actions.LISTING_CATEGORY_TVSHOWS)
         self._kodi_wrapper.show_listing(menu_items, sortmethod.ALPHABET)
 
-    def show_video_category_episodes(self, path):
-        category = path.split('/')[-2]
-        joined_url = self._VRTNU_SEARCH_URL + category
-        response = requests.get(joined_url)
-        programs = response.json()
-        menu_items = []
-        for program in programs:
-            title = program['title']
-            plot = BeautifulSoup(program['description'], 'html.parser').text
-            thumbnail = statichelper.replace_double_slashes_with_https(program['thumbnail'])
-
-            metadata_creator = metadatacreator.MetadataCreator()
-            metadata_creator.plot = plot
-            video_dictionary = metadata_creator.get_video_dictionary()
-            # cut vrtbase url off since it will be added again when searching for episodes (with a-z we dont have the full url)
-            link_to_video = program['targetUrl'].replace('//www.vrt.be', '')
-            item = helperobjects.TitleItem(title, {'action': actions.LISTING_VIDEOS, 'video': link_to_video},
-                                           False, thumbnail, video_dictionary)
-            menu_items.append(item)
-        self._kodi_wrapper.show_listing(menu_items, sortmethod.ALPHABET)
-
-    def play(self, url):
-        video = self._url_toStream_service.get_stream_from_url(url)
-        if video is not None:
-            self._kodi_wrapper.play(video)
+    def play(self, video):
+        stream = self._stream_service.get_stream(video)
+        if stream is not None:
+            self._kodi_wrapper.play(stream)
 
     def show_livestream_items(self):
         livestream_items = {
             helperobjects.TitleItem(self._kodi_wrapper.get_localized_string(32101),
-                                    {'action': actions.PLAY, 'video': self._EEN_LIVESTREAM},
-                                    True, self.__get_media('een.png')),
+                                    {'action': actions.PLAY, 'video_url': self._EEN_LIVESTREAM},
+                                    True, self._api_helper.get_live_screenshot('een')),
```

Comment:
It is nice to see a (recent) preview image when hovering the live channels, but the downside is that if you add the live channels to your favourites, it uses the preview image.

As an example I just (re)added the live channels to my favourites and Eén has an helicopter view of a city because of a cycling tournament, Canvas (actually Ketnet Jr) is now a screenshot of Uki and Ketnet has a Cartoon show in favourites. IIRC before these would just be the logo of the channel (brand).

I don't know if we can use the channel (brand) logo when it is being added to favourites, otherwise I am not in favor of this change.

Reply:
We cannot use registered trademark logos without permission of VRT. We can still use the "textual" representation images and make live preview images an addon setting.

Reply:
I wonder if we can have a handler that would return this "textual" representation when it is adding to favourites. That would be better than an option in preferences IMO. (Why bother users with this ?)

Reply:
(This is not a reason for me not merging this)

Reply:
I will move this item to a TODO list.